src/libs/ontouml2gufo/pre_analysis.ts

```diff
@@ -0,0 +1,368 @@
+    ...repeatedNameAnalysis,
+    ...inexistentCardinality,
+    ...inexistentAttributesType,
+  ];
+}
+
+async function checkBaseIRI(baseIRI: string): Promise<IPreAnalysisItem[]> {
+  const preAnalysis: IPreAnalysisItem[] = [];
+
+  if (!isURI(baseIRI)) {
+    preAnalysis.push({
+      id: randomId(),
+      code: 'invalid_base_iri',
+      title: 'Invalid BaseIRI',
+      detail: `"${baseIRI}" is not a valid IRI.`,
+      meta: { baseIRI },
+    });
+  }
+
+  return preAnalysis;
+}
+
+async function checkPackagePrefixes(
+  packages: IPackage[],
+  options: IOntoUML2GUFOOptions,
+): Promise<IPreAnalysisItem[]> {
+  const preAnalysis: IPreAnalysisItem[] = [];
+  const { customPackageMapping, prefixPackages } = options;
+  const defaultPrefixKeys = Object.keys(DefaultPrefixes);
+  const defaultPrefixUris = Object.values(DefaultPrefixes);
+
+  for (const key of Object.keys(customPackageMapping)) {
+    const packageEl =
+      packages.find(({ id, name }: IPackage) => id === key || name === key) ||
+      {};
+    const { prefix, uri } = customPackageMapping[key];
+
+    if (defaultPrefixKeys.includes(prefix)) {
+      preAnalysis.push({
+        id: randomId(),
+        code: 'invalid_custom_package_prefix',
+        title: 'Protected prefix provided in custom package mapping',
+        detail: `The prefix "${prefix}" is already used by another package imported by gUFO. Avoid using the following prefixes: ${defaultPrefixKeys.join(
+          ', ',
+        )}.`,
+        meta: { element: packageEl },
+      });
+    }
+
+    if (defaultPrefixUris.includes(uri)) {
+      preAnalysis.push({
+        id: randomId(),
+        code: 'invalid_custom_package_uri',
+        title: 'Protected URI provided in custom package mapping',
+        detail: `The URI "${uri}" is already used by another package imported by gUFO.`,
+        meta: { element: packageEl },
+      });
+    }
+  }
+
+  if (prefixPackages) {
+    const prefixes = await getPrefixes(packages, options);
+
+    for (const prefix of Object.keys(prefixes)) {
+      const uri = prefixes[prefix];
+
+      if (defaultPrefixKeys.includes(prefix)) {
+        preAnalysis.push({
+          id: randomId(),
+          code: 'invalid_package_prefix',
+          title: 'Protected prefix generated in package mapping',
+          detail: `The prefix "${prefix}" is already used by another package imported by gUFO. Beware of the following prefixes: ${defaultPrefixKeys.join(
+            ', ',
+          )}.`,
+          meta: { prefix, uri },
+        });
+      }
+
+      if (defaultPrefixUris.includes(uri)) {
+        preAnalysis.push({
+          id: randomId(),
+          code: 'invalid_package_uri',
+          title: 'Protected URI generated in package mapping',
+          detail: `The URI "${uri}" is already used by another package imported by gUFO.`,
+          meta: { prefix, uri },
+        });
+      }
+    }
+  }
+
+  return preAnalysis;
+}
+
+async function checkPluralAssociationEnd(
```

Comment:
I'm not sure we should keep this check. This pluralize package is not very precise in identifying plural nouns. You even use a verb instead of a plural noun in your example.

We may end up generating a bunch of annoying messages to the user.

Reply:
Removed